tasks/survey/minerprotocols/minerprotocols.go

```diff
@@ -147,9 +147,11 @@ func (t *Task) Close() error {
 func fetchMinerProtocolModel(ctx context.Context, api API, addr address.Address, minerInfo lapi.MinerInfo, start time.Time, results chan *observed.MinerProtocol) {
 	// since miners may choose if their peerID is set in their info
 	var peerID string
-	if minerInfo.PeerId != nil {
-		peerID = minerInfo.PeerId.String()
```

Comment:
I think this would be more clear if this check were moved into `getMinerAddrInfo`. If the peerID is nil that method can just return and log an error